File: billing-application/src/main/java/com/example/billing/application/port/in/IngestUsageUseCase.java
package com.example.billing.application.port.in;

import com.example.billing.application.command.IngestUsageCommand;

public interface IngestUsageUseCase {

    /**
     * @return true = 새로 저장됨, false = 이미 처리된 eventId (멱등성으로 무시)
     */
    boolean ingest(IngestUsageCommand cmd);
}

File: billing-application/src/main/kotlin/com/example/billing/application/port/in/IngestUsageUseCase.kt
package com.example.billing.application.port.`in`

import com.example.billing.application.command.IngestUsageCommand

interface IngestUsageUseCase {

    /**
     * @return true = 새로 저장됨, false = 이미 처리된 eventId (멱등성으로 무시)
     */
    fun ingest(cmd: IngestUsageCommand): Boolean
}
